### PR TITLE
Improve Max! devices discovery service

### PR DESCRIPTION
**Pull request by marcelrv**
*Improve loading & unloading of discovery service
*Improve behavour at dispose of a MAX! device

Signed-off-by: Marcel Verpaalen <marcel@verpaalen.com>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Thanks. Could you please apply the code formatter?

File: addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/discovery/MaxDeviceDiscoveryService.java [hunk 1]
```diff
@@ -91,7 +91,7 @@ public void onDeviceAdded(Bridge bridge, Device device) {
 
 	@Override
 	protected void startScan() {
-		// this can be ignored here as we discover via the bridge
+		if (maxCubeBridgeHandler != null) maxCubeBridgeHandler.clearDeviceList();
```
Comment (hakan42) [inline addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/discovery/MaxDeviceDiscoveryService.java]: Could you please add curly braces around the clearDeviceList() (event if it is just one call)? This makes debugging much easier.

**Comment by marcelrv** [comment]
@kaikreuzer okay done that

**Comment by kaikreuzer** [comment]
Hm, it does not see so - I still see many tabs in the code instead of spaces...

**Comment by marcelrv** [comment]
strange. Applied the formatting and it did seem to make changes.
Let me check my eclipse settings here.

**Comment by marcelrv** [comment]
@kai,
I see that indeed I had only the standard code formatter in my setup instead of the esh formatting config.

Suggest that I make separate pr with formatting update to the whole binding (to keep functionality update separate from the formatting change)

**Comment by kaikreuzer** [comment]
@marcelrv Sorry, I am a bit lost about the status of this PR. What I am seeing is that the formatting is not yet applied, but still the diff shows more or less all classes as completely changed. This is not what I expected...

**Comment by marcelrv** [comment]
@kaikreuzer then I'm also bit lost :( I re-installed the eclipse environment with new way to get the right code formatter (btw: huge improvement over the old method!)
The formatter indeed saw that whole classes needed to be updated, so I let it do that.

What you advice, run the formatter over the whole binding (it will see many many changes as seems my original eclipse formatter wanted tabs instead of spaces) or have a separate PR for that?

**Comment by kaikreuzer** [comment]
I would prefer to have two separate PRs - one with the code changes and one with the formatting on the whole binding - this makes it easier in the end to track changes.
The order of these 2 PRs is up to you :-)

**Comment by marcelrv** [comment]
@kaikreuzer so this is the one with the code changes. 
After this is merged I'll submit a PR with formatting changes with the right formatter for the whole binding

**Comment by kaikreuzer** [comment]
thanks!

**Pull request merged**